sqlflow/client.py

```diff
@@ -1,13 +1,50 @@
+import sys
 import os
 import logging
-
 import grpc
+
 import google.protobuf.wrappers_pb2 as wrapper
+from google.protobuf.timestamp_pb2 import Timestamp
 
 import sqlflow.proto.sqlflow_pb2 as pb
 import sqlflow.proto.sqlflow_pb2_grpc as pb_grpc
 
+
 _LOGGER = logging.getLogger(__name__)
+_OUT_HANDLER = logging.StreamHandler(sys.stdout)
+_OUT_HANDLER.setLevel(logging.INFO)
+_LOGGER.addHandler(_OUT_HANDLER)
+_LOGGER.setLevel(logging.INFO)
+
+
+class Rows:
+    def __init__(self, column_names, rows_gen):
+        self._column_names = column_names
+        self._rows_gen = rows_gen
+        self._rows = None
+
+    def column_names(self):
+        return self._column_names
+
+    def rows(self):
+        if self._rows is None:
+            self._rows = [r for r in self._rows_gen()]
+        return self._rows
```

Comment:
Try this:
for r in self._rows_gen():
  yield from r

Reply:
Do we need to support `rows()` to be called twice? how about

```python
def rows(self):
    if self._rows is None:
        self._rows = []
        for r in self._rows_gen():
            self._rows.append(r)
            yield r
    else:
        for r in self._rows:
            yield r
```

Reply:
cache it into _rows:)

Reply:
Done.